src/core/components/files-regular/refs-pull-stream.js

```diff
@@ -0,0 +1,115 @@
+'use strict'
+
+const exporter = require('ipfs-unixfs-exporter')
+const pull = require('pull-stream')
+const pullError = require('pull-stream/sources/error')
+const pullDefer = require('pull-defer')
+const { normalizePath } = require('./utils')
+const { Format } = require('./refs')
+
+module.exports = function (self) {
+  return function (ipfsPath, options = {}) {
+    if (options.maxDepth === 0) {
+      return pull.empty()
+    }
+    if (options.format !== Format.default && options.e) {
+      return pullError(Error('Cannot set edges to true and also specify format'))
+    }
+
+    options.format = options.e ? Format.edges : options.format || Format.default
+
+    const path = normalizePath(ipfsPath)
+    const pathComponents = path.split('/')
+
+    // eg QmHash/linkName => 2
+    const pathDepth = pathComponents.length
+
+    // The exporter returns a depth for each node, eg:
+    // Qmhash.../linkName/linkName/linkName/block
+    //    0         1         2        3      4
+    if (options.maxDepth === undefined) {
+      options.maxDepth = options.recursive ? global.Infinity : pathDepth
+    } else {
+      options.maxDepth = options.maxDepth + pathDepth - 1
+    }
+
+    if (options.preload !== false) {
+      self._preload(pathComponents[0])
+    }
+
+    // We need to collect all the values from the exporter and work out the
+    // parent of each node, so use a deferred source.
+    // TODO: It would be more efficient for the exporter to return the parent
+    // cid with the node, so we could just stream the result back to the
+    // client. Is this possible?
+    const deferred = pullDefer.source()
```

Comment:
@alanshaw it would be more efficient for the exporter to return the parent cid with the node, so we could just stream the result back to the client, instead of collecting all the links and working out the relationship between them. Do you know if this is feasible?